tests/filecheck/transformations/replace_op_test.mlir

```diff
@@ -9,11 +9,11 @@
 // CHECK: builtin.module {
 // CHECK:   "test.op"(%0) {replaced = "false"} : ("replaced(i32)") -> ()
 // CHECK:   "test.op"(%0) {replaced = "false"} : ("replaced(i32)") -> ()
-// CHECK:   "replacedOp"() ({
+// CHECK:   %0, %1 = "replacedOp"() ({
 // CHECK:     "dummy-op1"() : () -> ()
 // CHECK:     "dummy-op2"() : () -> ()
-// CHECK:     %0, %1 = "dummy-return"() : () -> ("replaced(i32)", "replaced(i64)")
-// CHECK:   }) : () -> ()
+// CHECK:     "dummy-return"() : () -> ()
+// CHECK:   }) : () -> ("replaced(i32)", "replaced(i64)")
 // CHECK:   "test.op"(%1) {replaced = "false"} : ("replaced(i64)") -> ()
 // CHECK:   "test.op"(%1) {replaced = "false"} : ("replaced(i64)") -> ()
 // CHECK: }
```

Comment:
I couldn't get this to work as it was!
I'm not sure it's doable at all; and to be clear, this previously violated a big assumption:
https://mlir.llvm.org/docs/LangRef/#value-scoping
> Similarly, regions provides a natural scoping for value visibility: values defined in a region don’t escape to the enclosing region, if any.

So I do think we should hold that - maybe we can provide more friendly checks for that? But this becomes a more technical requirement currently!